### PR TITLE
Fix for #1380 - Remove UIKit.h from Starboard.h

### PR DESCRIPTION
**Pull request by jaredhms**
I'm also fixing all warnings in UIKit and enabling warnings as errors.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1643)

### CODE REVIEW

File: Frameworks/CoreText/CTFontDescriptor.mm [hunk 1]
```diff
@@ -14,6 +14,9 @@
 //
 //******************************************************************************
 
+// TODO: File bug
```
Comment (bviglietta) [inline Frameworks/CoreText/CTFontDescriptor.mm]: >TODO: File bug [](start = 3, length = 14)

Have you filed this bug?

Comment (jaredhms) [inline Frameworks/CoreText/CTFontDescriptor.mm]: actually need to just remove the comment; CT already pulls in UIKit in a few other spots

---
In reply to: [94999743](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r94999743) [](ancestors = 94999743)

Comment (jaredhms) [inline Frameworks/CoreText/CTFontDescriptor.mm]: going to scrub my other 'todo file bug' comments too, before filing bugs.

---
In reply to: [95001484](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95001484) [](ancestors = 95001484,94999743)

File: Frameworks/include/Starboard.h [hunk 2]
```diff
@@ -35,8 +35,6 @@
 
 #ifdef __OBJC__
 
-#include "UIKit/UIKit.h"
```
Comment (yiyang-msft) [inline Frameworks/include/Starboard.h]: Yay!

File: Frameworks/UIKit/UIView.mm [hunk 3]
```diff
@@ -3481,7 +3503,7 @@ - (void)setGestureRecognizers:(NSArray*)gestures {
     [priv->gestures release];
```
Comment (bviglietta) [inline Frameworks/UIKit/UIView.mm]: >[priv->gestures release]; [](start = 4, length = 25)

gestures is a StrongId now.

File: Frameworks/include/WYPopoverController.h [hunk 4]
```diff
@@ -31,6 +31,10 @@
 
 #import "WYPopoverPrivate.h"
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnullability-completeness"
```
Comment (yiyang-msft) [inline Frameworks/include/WYPopoverController.h]: >-Wnullability-completeness [](start = 34, length = 26)

curious, what warning this turns off?

File: Frameworks/UIKit/UIView.mm (hunk 3 above)
Comment (jaredhms) [inline Frameworks/UIKit/UIView.mm]: thanks; surprising that it compiled...

---
In reply to: [95002606](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95002606) [](ancestors = 95002606)

File: Frameworks/UIKit/UIView.mm [hunk 5]
```diff
@@ -3481,7 +3503,7 @@ - (void)setGestureRecognizers:(NSArray*)gestures {
...
     priv->gestures = gestures;
```
Comment (jaredhms) [inline Frameworks/UIKit/UIView.mm]: >  priv->gestures = gestures; [](start = 2, length = 28)

yikes - should be .attach now

**Comment by bviglietta** [comment]
>             NSLayoutConstraint* constraint = [(id)constraints objectAtIndex:i];

This cast was weird before and is super weird now.

(The if statement is also curious.)

---
Refers to: Frameworks/UIKit/NSLayoutConstraint.mm:354 in 46331d9. [](commit_id = 46331d9a94df74f6ebbc3f32cdb55e562c597990, deletion_comment = False)

File: Frameworks/include/WYPopoverController.h (hunk 4 above)
Comment (jaredhms) [inline Frameworks/include/WYPopoverController.h]: that is the warning - you can look it up online.  i filed a bug: [https://github.com/Microsoft/WinObjC/issues/1645](https://github.com/Microsoft/WinObjC/issues/1645)

---
In reply to: [95002863](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95002863) [](ancestors = 95002863)

**Comment by jaredhms** [comment]
>             NSLayoutConstraint* constraint = [(id)constraints objectAtIndex:i];

ack - fixing...

---
In reply to: [270983731](https://github.com/Microsoft/WinObjC/pull/1643#issuecomment-270983731) [](ancestors = 270983731)

---
Refers to: Frameworks/UIKit/NSLayoutConstraint.mm:354 in 46331d9. [](commit_id = 46331d9a94df74f6ebbc3f32cdb55e562c597990, deletion_comment = False)

**Comment by yiyang-msft** [comment]
>     _toolBar = _toolBar;

existing, mind put a todo here to determine why we are doing this?

---
Refers to: Frameworks/UIKit/UINavigationController.mm:491 in 46331d9. [](commit_id = 46331d9a94df74f6ebbc3f32cdb55e562c597990, deletion_comment = False)

File: Frameworks/UIKit/UIPickerView.mm [hunk 6]
```diff
@@ -357,7 +363,7 @@ - (int)_selectedRow {
 }
 
 - (id)_cellSelectedUp:(id)cell {
```
Comment (yiyang-msft) [inline Frameworks/UIKit/UIPickerView.mm]: in UITableViewInternal.h, it defines 
- (void)_cellSelectedUp:(UITableViewCell*)cell;

File: Frameworks/UIKit/UISegment.h [hunk 7]
```diff
@@ -13,14 +13,16 @@
...
 @public
     id _delegate;
     idretain _title;
-    idretain _image;
-    idretain _selectedBackground, _highlightedBackground;
-    idretain _dividerImage;
+    UIImage* _image;
+    UIImage* _selectedBackground;
+    UIImage* _highlightedBackground;
+    UIImage* _dividerImage;
```
Comment (bviglietta) [inline Frameworks/UIKit/UISegment.h]: Should these be StrongId? Lifetime management doesn't look right now.

**Comment by bbowman** [comment]
libobjc2 is midified?

File: Frameworks/UIKit/UISegment.h (hunk 7 above)
Comment (jaredhms) [inline Frameworks/UIKit/UISegment.h]: looks like, thanks!

---
In reply to: [95006851](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95006851) [](ancestors = 95006851)

File: Frameworks/include/UIApplicationInternal.h [hunk 8]
```diff
@@ -31,7 +31,7 @@
 - (void)_sendResumeEvent;
 - (void)_sendToastActionReceivedEvent:(NSDictionary*)toastAction;
 - (void)_sendVoiceCommandReceivedEvent:(WMSSpeechRecognitionResult*)result;
-- (void)_sendProtocolReceivedEvent:(WFUri*)uri;
+- (void)_sendProtocolReceivedEvent:(WFUri*)protocolUri source:(NSString*)source;
```
Comment (bbowman) [inline Frameworks/include/UIApplicationInternal.h]: this compiled before?!!?!

Comment (jaredhms) [inline Frameworks/include/UIApplicationInternal.h]: yes, with a warning

---
In reply to: [95008850](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95008850) [](ancestors = 95008850)

File: Frameworks/UIKit/UIPickerView.mm (hunk 6 above)
Comment (jaredhms) [inline Frameworks/UIKit/UIPickerView.mm]: even better; thanks!

---
In reply to: [95006609](https://github.com/Microsoft/WinObjC/pull/1643#discussion_r95006609) [](ancestors = 95006609)

**Comment by jaredhms** [comment]
ugh; no, it shouldn't be.  why is this showing up?

---
In reply to: [270989968](https://github.com/Microsoft/WinObjC/pull/1643#issuecomment-270989968) [](ancestors = 270989968)

**Comment by bbowman** [comment]
:shipit:

File: Frameworks/UIKit/UIScrollView.mm [hunk 9]
```diff
@@ -582,7 +588,7 @@ - (void)_setupLoadedEventHandler {
             WXFrameworkElement* scrollContentPresenter = FindTemplateChild(strongSelf->_scrollViewer, @"ScrollContentPresenter");
             if (scrollContentPresenter) {
                 WXCGrid* parent = rt_dynamic_cast<WXCGrid>(scrollContentPresenter.parent);
-                [parent.children insertObject:self->_contentImage atIndex:0];
+                [parent.children insertObject:strongSelf->_contentImage atIndex:0];
```
Comment (yiyang-msft) [inline Frameworks/UIKit/UIScrollView.mm]: >strongSelf [](start = 46, length = 10)

thanks!

File: Frameworks/UIKit/StarboardXaml/UWPBackgroundTask.mm [hunk 10]
```diff
@@ -24,15 +24,20 @@
  */
 
 #import <Starboard.h>
-#import <Foundation/Foundation.h>
+
+#import <UIKit\UIApplication.h>
```
Comment (yiyang-msft) [inline Frameworks/UIKit/StarboardXaml/UWPBackgroundTask.mm]: >\ [](start = 14, length = 1)

nit: /

**Comment by yiyang-msft** [comment]
:shipit:

**Comment by bviglietta** [comment]
:shipit:

**Pull request merged**